system/src/main.cpp

```diff
@@ -292,10 +292,22 @@ void system_power_management_update()
         power.begin();
         power.setInputCurrentLimit(900);     // 900mA
         power.setChargeCurrent(0,0,0,0,0,0); // 512mA
+        static bool lowBattEventNotified = false; // Whether 'low_battery' event was generated already
+        static bool wasCharging = false; // Whether the battery was charging last time when this function was called
+        const uint8_t status = power.getSystemStatus();
+        const bool charging = (status >> 4) & 0x03;
+        if (!charging && wasCharging) { // Check if charging has been stopped
```

Comment:
Because it's more likely that a low battery event will be generated while the electron is not charging the battery, it seems like this line should be inverted to reset `lowBattEventNotified` when the system goes from not charging to charging.
`if (charging && !wasCharging) {`.  Currently it would require two actions, a transition from !charging to charging, and then back to !charging.

Without this, it will not be possible to generate a low battery event while applying a heavy load under light/normal charging of the battery.